Character.cpp

```diff
@@ -68,26 +68,28 @@ bool Character::IsDead() const
 	return hp == 0;
 }
 
-void Character::DMGTaken(Character* foe)
+void Character::DMGTaken(Character*& foe)
 {
 	hp -= foe->getDmg();
 	if (hp <= 0) { hp = 0; }
 }
 
-void Character::OnePunch(Character* foe)
+void Character::OnePunch(Character*& foe)
 {
-	foe->DMGTaken(this);
+	Character* CharacterPointer = this;
+	foe->DMGTaken(CharacterPointer);
 }
 
-void Character::Fight(Character* foe)
+void Character::Fight(Character*& foe)
 {
 	int n = 0;
 	while (foe->getHp() != 0 && this->getHp() != 0) {
 		if (n % 2 == 0) {
 			this->OnePunch(foe);
 		}
 		else {
-			foe->OnePunch(this);
+			Character* CharacterPointer = this;
+			foe->OnePunch(CharacterPointer);
```

Comment:
Itt is látszik, hogy ez nem jó megoldás, azért nem fogadta el a `this` átadását, mert azt nem lehet módosítható referenciaként átadni.